src/main/java/seedu/planner/model/ModelManager.java

```diff
@@ -38,15 +53,44 @@ public ModelManager(ReadOnlyAddressBook addressBook, UserPrefs userPrefs) {
 
         versionedAddressBook = new VersionedAddressBook(addressBook);
         filteredPersons = new FilteredList<>(versionedAddressBook.getPersonList());
-        //TODO: initialise filteredModules properly
-        filteredTakenModules = new FilteredList<>(SampleModulesUtil.genModules(3));
-        filteredAvailableModules = new FilteredList<>(SampleModulesUtil.genModules(2));
+
+        //@@author GabrielYik
+
+        //TODO: initialise ModulePlanner properly
+        ModulePlanner modulePlanner = SampleModulePlannerUtil.genModulePlanner();
+        versionedModulePlanner = new VersionedModulePlanner(modulePlanner);
+        filteredTakenModules = setUpFilteredTakenModules();
```

Comment:
Why do we need to retrieve all list of modules here? Can't we just retrieve a list of modules for a specific semester when needed?

Reply:
I'm initialising all Semesters in ModulePlanner at the start in my PR, not sure we need to retrieve all modules lists from the start though.

Reply:
But won't initialising all semesters involve initialising all module lists too? Since a module list is in a semester

Reply:
Yeah, but the modules lists are still in their own Semester class, while what you are doing here is retrieving all modules lists and storing them in ModelManager

Reply:
Ohh. Do you mean I do something like this?
`List<Semester> semesters = versionedModulePlanner.getSemesters(int index);`
Then in the`ModelManager#getFilteredTakenModuleList`, I just get the correct semester using the `index` and retrieve the module list from that semester?

Reply:
Ok changed